pkg/providers/ssm/provider.go

```diff
@@ -60,3 +74,54 @@ func (p *DefaultProvider) Get(ctx context.Context, parameter string) (string, er
 	log.FromContext(ctx).WithValues("parameter", parameter, "value", result.Parameter.Value).Info("discovered ssm parameter")
 	return lo.FromPtr(result.Parameter.Value), nil
 }
+
+// Invalidate asyncronously invalidates entries in the underlying SSM cache. Invalidation is scheduled up to 30 minutes
+// in the future to avoid thundering herds caused by globally correlated invalidations. Any subsequent invalidations of
+// the same parameter made before the initial invalidation has been processed are no-ops.
+func (p *DefaultProvider) Invalidate(ctx context.Context, parameters ...string) {
+	p.Lock()
+	defer p.Unlock()
+
+	nonPending := map[string]string{}
+	for _, parameter := range parameters {
```

Comment:
We discussed this offline but if you are going this route consider using a workqueue so that you get dirty-set tracking and metrics for free